rubix-tests/src/test/java/com/qubole/rubix/tests/TestThriftServerJVM.java

```diff
@@ -0,0 +1,176 @@
+
+  public BookKeeperFactory bookKeeperFactory = new BookKeeperFactory();
+  public RetryingBookkeeperClient client;
+
+  static Process bookKeeperJvm;
+  static Process localDataTransferJvm;
+
+  private static Configuration conf = new Configuration();
+  private static String cacheDir = CacheConfig.getCacheDirPrefixList(conf) + "0" + CacheConfig.getCacheDataDirSuffix(conf);
+
+  @BeforeClass
+  public static void setupClass() throws IOException, InterruptedException
+  {
+    /*
+     * Dynamically, retrieving bookkeeper jar from /usr/lib/hadoop2/share/hadoop/tools/lib/ folder
+     * */
+
+    File folder = new File(jarsPath);
+    File[] listOfFiles = folder.listFiles();
+    String bookKeeperJarPath = null;
+    for (int i = 0; i < listOfFiles.length; i++) {
+      if (listOfFiles[i].isFile() && listOfFiles[i].toString().contains("bookkeeper")) {
+        bookKeeperJarPath = listOfFiles[i].toString();
+      }
+    }
+    log.debug(" Located Bookkeeper Jar is : " + bookKeeperJarPath);
+    /*
+     * Spinning up the separate JVMs for bookKeeper and Local Data Transfer
+     * */
+    String[] bookKeeperStartCmd = {hadoopDirectory, "jar", bookKeeperJarPath, bookKeeperClass, setDataBlockSize, setCacheMaxDisks};
+    String[] localDataTransferStartCmd = {hadoopDirectory, "jar", bookKeeperJarPath, localDataTransferServerClass, setDataBlockSize, setCacheMaxDisks};
+
+    ProcessBuilder pJVMBuilder = new ProcessBuilder();
+    pJVMBuilder.redirectErrorStream(true);
+    pJVMBuilder.inheritIO();
+    pJVMBuilder.command(bookKeeperStartCmd);
+    bookKeeperJvm = pJVMBuilder.start();
+    pJVMBuilder.command(localDataTransferStartCmd);
+    localDataTransferJvm = pJVMBuilder.start();
+
+    log.info("Test Directory : " + testDirectory);
+    Files.createDirectories(Paths.get(testDirectory));
+    Thread.sleep(3000);
+  }
+
+  @AfterClass
+  public static void tearDownClass() throws IOException
+  {
+    /* ****
+      Destroying bookKeeper and localDataTransfer JVMs
+    **** */
+    bookKeeperJvm.destroy();
+    localDataTransferJvm.destroy();
+
+    log.info("Deleting files in " + testDirectory);
+    Files.walkFileTree(Paths.get(testDirectory), new DeleteFileVisitor());
+    Files.deleteIfExists(Paths.get(testDirectory));
+
+    Files.walkFileTree(Paths.get(cacheDir + "/" + testDirectoryPrefix), new DeleteFileVisitor());
+    Files.deleteIfExists(Paths.get(cacheDir + "/" + testDirectoryPrefix));
+  }
+
+  @BeforeMethod
+  public static void setup() throws IOException, InterruptedException, URISyntaxException
+  {
+    DataGen.populateFile(backendFileName);
+    log.info("BackendPath: " + backendPath);
+  }
+
+  @AfterMethod
+  public static void cleanup()
+  {
+    File file = new File(backendFileName);
+    file.delete();
+  }
+
+  @Test
+  public void testJVMCommunication() throws IOException, InterruptedException
+  {
+    log.info("Value of Path " + this.backendFileName);
```

Comment:
Seems like there is too much setup here. Does some of this have to move to a setup method ?

Reply:
I agree we should move make conf as static variable and set some of these configs in beforeclass or beforemethod

Reply:
I will move some of the initialization code to setup method